packages/dustkit/src/apps/getProgramAppConfigUrl.ts

```diff
@@ -18,17 +18,30 @@ export async function getProgramAppConfigUrl({
   program: ProgramId;
   entity?: EntityId;
 }): Promise<string | undefined> {
-  const url = await readContract(client, {
+  let url: string | undefined;
+  const { system } = await getRecord(client, {
     address: worldAddress,
-    abi: worldCallAbi,
-    functionName: "call",
-    args: encodeSystemCall({
+    table: baseWorldConfig.tables.world__Systems,
+    key: {
       systemId: program,
+    },
+  });
+
+  try {
+    // We are calling the program system directly here, as programs are required to be private
+    // systems so that their hook functions can only be called by the world
+    url = await readContract(client, {
+      address: system,
       abi: appAbi,
       functionName: "appConfigURI",
       args: [entity ?? zeroHash],
-    }),
-  });
+    });
+  } catch (error: any) {
```

Comment:
there's examples in mud for how to do this better than with any

Reply:
updated